tests/test_database.py

```diff
@@ -95,6 +96,56 @@ def test_write_session(example_database: Database, example_subject: Subject):
     reloaded_session = example_database.load_session(example_subject, session.id)
     assert reloaded_session.name == "new_name"
 
+def test_write_run(example_database: Database, tmp_path:Path):
+    subject_id = "example_subject"
+    session_id = "example_session"
+    protocol_id = "example_protocol"
+    run_id = "example_run_2"
+    success_flag = True
+    note = "Test note"
+    solution_id = "example_solution"
+    subject = example_database.load_subject(subject_id)
+    session = example_database.load_session(subject, session_id)
+    protocol = example_database.load_protocol(protocol_id)
+    run = Run(id=run_id, success_flag=success_flag, note=note, session_id=session_id, solution_id=solution_id)
+
+    # Can add a new run
+    example_database.write_run(run, session, protocol)
+    run_file_path = tmp_path/"example_db/subjects/example_subject/sessions/example_session/runs/example_run/example_run.json"
+    assert(run_file_path.is_file())
+
+    # Error raised when the session already exists
+    with pytest.raises(ValueError, match="already exists"):
+        example_database.write_run(run, session, protocol, on_conflict=OnConflictOpts.ERROR)
+
+    # Error raised when the user try to overwrite a run
+    with pytest.raises(ValueError, match="may not be overwritten"):
+        example_database.write_run(run, session, protocol, on_conflict=OnConflictOpts.OVERWRITE)
+
+    # Make sure the Runs folder and Runs file are created
+    new_session = Session(id='new_session',subject_id='example_subject')
+    example_database.write_session(subject, new_session)
+    new_run = Run(id=run_id, success_flag=success_flag, note=note, session_id='new_session', solution_id=solution_id)
+    example_database.write_run(new_run, new_session, protocol, on_conflict=OnConflictOpts.OVERWRITE)
+    runs_filename = example_database.get_runs_filename(subject.id, new_session.id)
+    assert runs_filename.exists()
+    assert runs_filename.is_file()
+    assert runs_filename.name == "runs.json"
```

Comment:
@ebrahimebrahim  before i merge, i noticed that it could be useful to add this test that make sure that Runs file is created when a run is written for the first time for a particular session